api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java

```diff
@@ -387,18 +406,181 @@ public boolean supportsPropertyName(String gpName) {
 	}
 
 	@Override
-	public String importMpiPatient(String personId) {
-		if (registrationCoreProperties.isMpiEnabled()) {
-			MpiProvider mpiProvider = registrationCoreProperties.getMpiProvider();
-			Patient importedPatient = mpiProvider.fetchMpiPatient(personId);
-			Patient patient = patientService.savePatient(importedPatient);
-			return patient.getUuid();
+	public MpiPatient findMpiPatient(String identifier, String identifierTypeUuid) {
+		if (!registrationCoreProperties.isMpiEnabled()) {
+			throw new MpiException("Should not perform 'findMpiPatient' when MPI is disabled");
+		}
+		MpiProvider mpiProvider = registrationCoreProperties.getMpiProvider();
+		MpiPatient mpiPatient = null;
+		try {
+			mpiPatient = mpiProvider.fetchMpiPatient(identifier, identifierTypeUuid);
+		} catch (RuntimeException e) {
+			servePdqExceptionAndThrowAgain(e,
+					prepareParameters(identifier, identifierTypeUuid),
+					PdqErrorHandlingService.FIND_MPI_PATIENT_DESTINATION);
+		}
+		return mpiPatient;
+		//TODO throw error if you get more than one patient
+	}
+
+	@Override
+	public Patient importMpiPatient(String personId) {
+		return importMpiPatient(personId, mpiProperties.getMpiPersonIdentifierTypeUuid());
+	}
+
+	@Override
+	public Patient importMpiPatient(String patientIdentifier, String patientIdentifierTypeUuid) {
```

Comment:
I think it would be more intuitive to name this method `importPatientFromMpi`

Reply:
Mmm I think that importMpiPatient is intuitive enough... if we change it to importPatientFromMpi, we also have to change the places that call it in registrationapp. 

Also, then we should change:
fetchMpiPatient -> fetchPatientFromMpi
updatePatient -> updatePatientInMpi
exportPatient -> exportPatientToMpi
findMpiPatient -> findPatientInMpi
persistImportedMpiPatient -> persistPatientImportedFromMpi

for consistency 
As well as change all the tests...
Which is doable! I personally think it is intuitive/readable/concise enough, but thats just my personal opinion. If you think that its really much more intuitive/readable I can definitely do it.